tests/models/test_baseoperator.py

```diff
@@ -347,3 +350,67 @@ def test_lineage_composition(self):
         task4 = DummyOperator(task_id="op4", dag=dag)
         task4 > [inlet, outlet, extra]
         self.assertEqual(task4.get_outlet_defs(), [inlet, outlet, extra])
+
+
+@pytest.fixture(scope="class")
+def provide_test_dag_bag():
+    yield DagBag(dag_folder=TEST_DAGS_FOLDER, include_examples=False)
+
+
+class TestXComArgsRelationsAreResolved:
+    def test_upstream_is_set_when_template_field_is_xcomarg(
+        self, provide_test_dag_bag  # pylint: disable=redefined-outer-name
+    ):
```

Comment:
I have more commonly seen this done as:

```suggestion
@pytest.fixture(scope="class")
def test_dag_bag():
    return DagBag(dag_folder=TEST_DAGS_FOLDER, include_examples=False)


class TestXComArgsRelationsAreResolved:
    def test_upstream_is_set_when_template_field_is_xcomarg(
        self, test_dag_bag  # pylint: disable=redefined-outer-name
    ):
```

No need for yield as we don't do any tidy up is there?

And `provide_` isn't how Pytest suggest doing it in their docs https://docs.pytest.org/en/latest/fixture.html#fixtures-as-function-arguments -- so in the case where we _use_ the value, I'd like us to just have `test_dag_bag` for example.

Reply:
Right, thanks for checking this